osbuild/api.py

```diff
@@ -193,6 +201,18 @@ def _cleanup(self):
             os.close(self._output_pipe)
             self._output_pipe = None
 
+def exception(path="/run/osbuild/api/osbuild"):
+    """Send exception to osbuild"""
+    _, exc_value, _ = sys.exc_info()
+    traceback.print_exc()
+    with jsoncomm.Socket.new_client(path) as client:
+        msg = {
+            "method": "exception",
+            "traceback": str(exc_value)
+        }
+        client.send(msg)
+
+    sys.exit(2)
```

Comment:
How about adding an additional helper like this:

```python
@contextlib.contextmanager
def exception_handler(path="..."):
    try:
        yield
    except Exception as e:
        exception(e)
```

This simplifies the callside, and we can do:

```python
with exception_handler():
    ...code...
```

Reply:
Makes sense, thanks - what would be an appropriate exception(s) to catch here?